assembly/src/parsers/io_ops.rs

```diff
@@ -17,16 +17,36 @@ const HEX_CHUNK_SIZE: usize = 16;
 // INSTRUCTION PARSERS
 // ================================================================================================
 
-/// Returns `PushConstants` instruction node.
+/// Returns one of the `Push` instruction nodes.
 ///
 /// # Errors
 /// Returns an error if the instruction token has invalid values or inappropriate number of
 /// values.
 pub fn parse_push(op: &Token) -> Result<Node, ParsingError> {
-    validate_operation!(op, "push", 1..MAX_PUSH_INPUTS);
-
-    let constants = parse_constants(op)?;
-    Ok(Instruction(PushConstants(constants)))
+    debug_assert_eq!(op.parts()[0], "push");
+    match op.num_parts() {
+        0 => unreachable!(),
+        1 => Err(ParsingError::missing_param(op)),
+        2 => {
+            let param_str = op.parts()[1];
+            if let Some(param_str) = param_str.strip_prefix("0x") {
+                // if we have only one parameter
+                if param_str.len() <= HEX_CHUNK_SIZE {
+                    // check parameter validity
+                    if param_str.len() % 2 != 0 {
+                        return Err(ParsingError::invalid_param(op, 1));
+                    }
```

Comment:
I found a strange thing -- previously we weren't checking that a hex string is of even length while parsing push with multiple parameters, e.g. `push.0xA.0x64.0x3E8.0x2710.0x186A0` was parsing with `u64::from_str_radix()` without this check (and without check that a string is shorter than 16 characters, I think we need to add it) and it was working fine. 
Should we add this requirement to every hex string, or let single hex parameter and multiple hex parameters with delimiter be parsed without this check?

Reply:
I would still have this check because in some other languages this is a requirement, and it would be good to have Miden assembly be more portable.

So, we should have this check in all cases, and should probably return an error which has a better error message.